src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -33,53 +33,208 @@ EndContentData */
 
 enum
 {
-    SAY_PEON_AWAKE_1        = -1000795,
-    SAY_PEON_AWAKE_2        = -1000796,
+    SAY_PEON_AWOKEN = -1000795,
 
-    SPELL_PEON_SLEEP        = 17743,
-    SPELL_AWAKEN_PEON       = 19938,
+    SOUND_PEON_WORK = 6197,
+    SOUND_PEON_AWAKE_1 = 6292,
+    SOUND_PEON_AWAKE_2 = 6294,
 
-    NPC_SLEEPING_PEON       = 10556,
-    GO_LUMBERPILE           = 175784,
+    SPELL_PEON_SLEEP = 17743,
+    SPELL_AWAKEN_PEON = 19938,
+
+    NPC_SLEEPING_PEON = 10556,
+    GO_LUMBERPILE = 175784,
 };
 
 struct npc_lazy_peonAI : public ScriptedAI
 {
     npc_lazy_peonAI(Creature* pCreature) : ScriptedAI(pCreature)
     {
         Reset();
-        m_uiStopSleepingTimer = urand(30000, 120000);       // Set on spawn to a potential small timer, to get nice results for initial case
+        //tree waypoint
+        //TODO do this via invisible creatures once they're added in the db (or actual waypoints)
+        switch (m_creature->GetGUIDLow())
+        {
```

Comment:
**OMG!!! Using GUIDLow in SD2?!? This is clearly a NO NO.**

We must to find another way. For example creature_movement, which can support GUID movement and positions.
GUIDs are totally controlled by DB, so we cannot force a DB provider to set these GUIDs for the NPCs just because we want to.